packages/testing/examples/useUniswapPrice.test.tsx

```diff
@@ -0,0 +1,76 @@
+import { MockProvider } from '@ethereum-waffle/provider'
+import { solidity } from 'ethereum-waffle'
+import { Contract } from '@ethersproject/contracts'
+import { pack, keccak256 } from '@ethersproject/solidity'
+import { getCreate2Address } from '@ethersproject/address'
+import { FixedNumber } from '@ethersproject/bignumber'
+import chai, { expect } from 'chai'
+import { useUniswapPrice, INIT_CODE_HASH, UniswapV2Pair } from '@usedapp/uniswap'
+import { compareAddress } from '@usedapp/core'
+import { renderWeb3Hook } from '../src'
+import { deployMockToken, MOCK_TOKEN_INITIAL_BALANCE } from '../src/utils/deployMockToken'
+import { deployUniswapV2Pair } from '../src/utils/deployMockUniswapV2Pair'
+import { BigNumber } from 'ethers'
+
+chai.use(solidity)
+
+describe('useUniswapPrice', () => {
+  const mockProvider = new MockProvider()
+  const [deployer] = mockProvider.getWallets()
+  const DIGITS = 8
+  const ONE = BigNumber.from(1)
+  const RATIO = BigNumber.from(5)
+  const EXP_SCALE = BigNumber.from(10).pow(DIGITS)
+  let tokenA: Contract
+  let tokenB: Contract
+  let factory: Contract
+  let pair: Contract
+
+  async function addLiquidity(tokenAAmount: BigNumber, tokenBAmount: BigNumber) {
+    await tokenA.transfer(pair.address, tokenAAmount)
+    await tokenB.transfer(pair.address, tokenBAmount)
+    await pair.mint(deployer.address)
+  }
+
+  function sortAddress(tokenA: string, tokenB: string) {
+    return compareAddress(tokenA, tokenB) === -1 ? [tokenA, tokenB] : [tokenB, tokenA]
+  }
+
+  beforeEach(async () => {
+    tokenA = await deployMockToken(deployer)
+    tokenB = await deployMockToken(deployer)
+    ;({ factory, pair } = await deployUniswapV2Pair(deployer, tokenA, tokenB))
```

Comment:
There's a semicolon here, should it be there?

Reply:
First, I think so too, but It seems to require a semicolon. If there's  no semicolon, compilation will fail.
``` 
This expression is not callable.
  Type 'Promise<Contract>' has no call signatures.

41     tokenB = await deployMockToken(deployer)
                      ~~~~~~~~~~~~~~~~~~~~~~~~~

  examples/useUniswapPrice.test.tsx:41:20
    41     tokenB = await deployMockToken(deployer)
                          ~~~~~~~~~~~~~~~~~~~~~~~~~
    Are you missing a semicolon?
```

MDN's js reference says about assignment without declaration
> The parentheses ( ... ) around the assignment statement are required when using object literal destructuring assignment without a declaration. ..
...
Your ( ... ) expression needs to be preceded by a semicolon or it may be used to execute a function on the previous line.